### PR TITLE
drivers/rn2xx3 : Expose Configurations to Kconfig

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in RN2XX3 Network Device driver to Kconfig.

### Testing procedure

New macro was introduced in tests/driver_rn2xx3/main.c for testing.

```
#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))
```
Firmware was uploaded to FIT IoT Lab test bed.

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-199-gd936bb-Kconfig_rn2xx3)
CONFIG_RN2XX3_DEFAULT_SLEEP=(5000U)
RN2XX3 device driver test
RN2XX3 initialization failed

#### Usage with CFLAGS 

/tests/driver_rn2xx3/Makefile

> CFLAGS += -DCONFIG_RN2XX3_DEFAULT_SLEEP=7000

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-199-gd936bb-Kconfig_rn2xx3)
CONFIG_RN2XX3_DEFAULT_SLEEP=7000
RN2XX3 device driver test
RN2XX3 initialization failed

#### Usage with Kconfig

/tests/driver_rn2xx3/

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-199-gd936bb-Kconfig_rn2xx3)
CONFIG_RN2XX3_DEFAULT_SLEEP=10000
RN2XX3 device driver test
RN2XX3 initialization failed
Note : The sensor is not available for interfacing hence configurability of macros were only tested.

### Issues/PRs references

#12888
@leandrolanzieri

### CODE REVIEW

File: drivers/rn2xx3/Kconfig [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+if KCONFIG_MODULE_RN2XX3
+
+config RN2XX3_DEFAULT_SLEEP
+    int "Sleep Duration (ms)"
+    default 5000
+    help
+        Set the sleep duration (in ms).
+        The value should be greater than RN2XX3_SLEEP_MIN (100 ms)
+
+
```
Comment (leandrolanzieri) [inline drivers/rn2xx3/Kconfig:22]: New line not needed

File: drivers/rn2xx3/Kconfig [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        Configure the RN2XX3 driver using Kconfig.
+
+if KCONFIG_MODULE_RN2XX3
+
+config RN2XX3_DEFAULT_SLEEP
+    int "Sleep Duration (ms)"
+    default 5000
+    help
+        Set the sleep duration (in ms).
+        The value should be greater than RN2XX3_SLEEP_MIN (100 ms)
```
Comment (leandrolanzieri) [inline drivers/rn2xx3/Kconfig:20]: If this is the case then you can set a range restriction to the parameter

File: drivers/rn2xx3/Kconfig [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+menuconfig KCONFIG_MODULE_RN2XX3
+    bool "Configure RN2XX3 driver"
+    depends on MODULE_RN2XX3
+    help
+        Configure the RN2XX3 driver using Kconfig.
+
+if KCONFIG_MODULE_RN2XX3
+
+config RN2XX3_DEFAULT_SLEEP
+    int "Sleep Duration (ms)"
```
Comment (leandrolanzieri) [inline drivers/rn2xx3/Kconfig:16]:
```suggestion
    int "Sleep duration in milliseconds [ms]"
```

File: drivers/rn2xx3/Kconfig (hunk 1 above)
Comment (akshaim) [reply drivers/rn2xx3/Kconfig:22]: Oops.

File: drivers/rn2xx3/Kconfig (hunk 2 above)
Comment (akshaim) [reply drivers/rn2xx3/Kconfig:23]: Done. Created Macros for the values.

File: drivers/rn2xx3/Kconfig [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+source "$(RIOTBASE)/Kconfig.macros"
```
Comment (leandrolanzieri) [inline drivers/rn2xx3/Kconfig:7]: If we keep the `Kconfig.macros` file, this should be `source`d in the root Kconfig

Comment (akshaim) [reply drivers/rn2xx3/Kconfig]: Done. Updated [b63bc8d ](https://github.com/RIOT-OS/RIOT/pull/13952/commits/b63bc8d4dda1c2d30fab03bdbcab147b43e7318a)

File: Kconfig.macros [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
```
Comment (leandrolanzieri) [inline Kconfig.macros:1]: I think it's better if we create a folder and move this file to `/kconfig/Kconfig.consts`

Comment (akshaim) [reply Kconfig.macros]: Done.

File: Kconfig [hunk 6]
```diff
@@ -6,6 +6,9 @@
 #
 mainmenu "RIOT Configuration"
 
+# Load macro definitions
+source "$(RIOTBASE)/kconfig/Kconfig.consts"
```
Comment (leandrolanzieri) [inline Kconfig:10]:
```suggestion
rsource "kconfig/Kconfig.consts"
```

Comment (akshaim) [reply Kconfig]: Thanks. Rebased and tested.

**Pull request merged**